### PR TITLE
Network Topology Tables as ValueMaps 

### PR DESCRIPTION
**Pull request by aeliot**
Added support for getting the topology tables (neighbor table and child table) in ValueMap format rather then string format.

### CODE REVIEW

File: src/ncp-spinel/SpinelNCPInstance.cpp [hunk 1]
```diff
@@ -490,6 +499,16 @@ SpinelNCPInstance::get_property(
 			)
 		));
 
+		} else if (strcaseequal(key.c_str(), kWPANTUNDProperty_ThreadNeighborTableAsValMap)) {
```
Comment (abtink) [inline src/ncp-spinel/SpinelNCPInstance.cpp]: nit: spacing alignment (probably tabs vs spaces)

File: src/ncp-spinel/SpinelNCPTaskGetNetworkTopology.cpp [hunk 2]
```diff
@@ -333,6 +333,36 @@ SpinelNCPTaskGetNetworkTopology::TableEntry::get_as_string(void) const
...
+    addr |= (uint64_t) mExtAddress[6] << 8;
+    addr |= (uint64_t) mExtAddress[5] << 16;
+    addr |= (uint64_t) mExtAddress[4] << 24;
+    addr |= (uint64_t) mExtAddress[3] << 32;
+    addr |= (uint64_t) mExtAddress[2] << 40;
+    addr |= (uint64_t) mExtAddress[1] << 48;
+    addr |= (uint64_t) mExtAddress[0] << 56;
+
+    entryMap.insert( std::pair<std::string, boost::any>( "extAddress",     addr               ));
+    entryMap.insert( std::pair<std::string, boost::any>( "rolc16",         mRloc16            ));
```
Comment (abtink) [inline src/ncp-spinel/SpinelNCPTaskGetNetworkTopology.cpp]: I suggest that we define all these strings in a header file somewhere (maybe in wpan-properties), something like this: 
```
#define kWPANTUNDValueMapKey_ChildNeighborEntry_ExtAddress        "ExtAddress"
#define kWPANTUNDValueMapKey_ChildNeighborEntry_Rloc16            "RLOC16" 
#define kWPANTUNDValueMapKey_ChildNeighborEntry_LinkQualityIn     "LinkQualityIn"
...
```
This way there is one place they are defined and the same `#define`s can later be used in sapphire code.

Comment (aeliot) [inline src/ncp-spinel/SpinelNCPTaskGetNetworkTopology.cpp]: Do you think the get_as_string function above should also use the same names?  The keys are currently the same as the names used when outputting as a string

Comment (abtink) [inline src/ncp-spinel/SpinelNCPTaskGetNetworkTopology.cpp]: Maybe... would be good to change it but not critical...

**Pull request merged**